main.cpp

```diff
@@ -73,43 +119,104 @@ bool rentACar(int rentalDuration, int carType = 0)  //function declaration with
 
 /*
  1)
- */
+ */
+int guessTheNumber(int theNumber, int numberOfTries = 3)
+{
+    ignoreUnused(theNumber, numberOfTries);
+    return {};
+}
 
 /*
  2)
- */
+ */
+
+ float doSomeMath(int smallNumber, int bigNumber, float coursePi = 3.14f)
+ {
+     ignoreUnused(smallNumber, bigNumber, coursePi);
+     return {};
+ }
 
 /*
  3)
- */
+ */
+
+ bool checkTheStatus(bool currentStatus)
+ {
+     ignoreUnused(currentStatus);
+     return {};
+ }
 
 /*
  4)
- */
+ */
+
+ void questionExistence(int age, int midlifeCrisis = 40)
+ {
+     ignoreUnused(age, midlifeCrisis);
+ }
 
 /*
  5)
- */
+ */
+
+ double calculateTheThing(float thing1, float thing2, float thing3)
+ {
+     ignoreUnused(thing1, thing2, thing3);
+     return {};
+ }
 
 /*
  6)
- */
+ */
+
+ int basicAdding(int number1, int number2)
+ {
+    int sum = number1 + number2;
+    return sum;
+ }
 
 /*
  7)
- */
+ */
+ bool greaterThan(int value1, int value2 = 0)
+ {
+     if (value1 > value2)
+     {
+         return true;
+     }
+     else
+     {
+         return false;
+     }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise